### PR TITLE
Addition of ```sectionStartDate``` field and field resolver.

### PR DESCRIPTION
**Pull request by Shinsina**
Screen Shot 2022-02-15 at 10 57 11 AM

### CODE REVIEW

**Comment by Shinsina** [comment]
[Update GraphQL API to return current section scheduled date](https://trello.com/c/HgTodiyh/280-update-graphql-api-to-return-current-section-scheduled-date)

File: services/graphql-server/src/graphql/resolvers/platform/content.js [hunk 1]
```diff
@@ -444,6 +444,26 @@ module.exports = {
       return isArray(sectionQuery) ? sectionQuery : [];
     },
 
+    /**
+     * Return the oldest start date for the current section
+     */
+    sectionStartDate: async (content, { input }, { basedb }) => {
+      const { sectionQuery } = content;
+      const { sectionId } = input;
+      const descendantIds = await getDescendantIds(sectionId, basedb);
```
Comment (solocommand) [inline services/graphql-server/src/graphql/resolvers/platform/content.js]: This should only happen if section bubbling is enabled. Add that to the input and only get the descendant ids in that case. Check the `websiteScheduledContent` query, match the input key and default value here as well (I believe on by default, but double check please!)

The rest of your logic should stay the same...something like `ids = useSectionBubbling ? await getDescendantIds() : [sectionId];`

File: services/graphql-server/src/graphql/resolvers/platform/content.js [hunk 2]
```diff
@@ -444,6 +444,26 @@ module.exports = {
...
+      const descendantSectionSchedules = sectionQuery
+        .filter(section => descendantIds.includes(section.sectionId))
+        .map(section => section.start);
+      const sectionDates = [
+        ...new Set([
+          ...sectionSchedules.map(section => section.start),
+          ...descendantSectionSchedules,
+        ]),
+      ];
+      return sectionDates[0];
```
Comment (solocommand) [inline services/graphql-server/src/graphql/resolvers/platform/content.js]: We don't care about uniqueness, you should just be able to do something like this
```js
  return schedules.length ? schedules[0].start : null;
```

File: services/graphql-server/src/graphql/definitions/platform/content/interfaces/content.js [hunk 3]
```diff
@@ -88,6 +88,8 @@ interface Content @requiresProject(fields: ["type"]) {
 
   # Allows accessing the content's taxonomy IDs without querying the related models.
   taxonomyIds: [Int!]! @projection(localField: "taxonomy")
+
```
Comment (solocommand) [inline services/graphql-server/src/graphql/definitions/platform/content/interfaces/content.js:91]: Add a GraphQL doc line above this field to indicate the expected use.

File: services/graphql-server/src/graphql/definitions/platform/content/index.js [hunk 4]
```diff
@@ -778,6 +778,10 @@ input ContentHasWebsiteScheduleInput {
   sectionBubbling: Boolean = true
 }
 
+input SectionStartDateInput {
+  sectionId: Int
```
Comment (solocommand) [inline services/graphql-server/src/graphql/definitions/platform/content/index.js]: Add GraphQL docs here to indicate what the input is used for and what each field (including the new flag) does

**Review by solocommand — CHANGES_REQUESTED** [review]

File: services/graphql-server/src/graphql/resolvers/platform/content.js [hunk 5]
```diff
@@ -444,6 +444,26 @@ module.exports = {
...
+     * Return the oldest start date for the current section
+     */
+    sectionStartDate: async (content, { input }, { basedb }) => {
+      const { sectionQuery } = content;
+      const { sectionId } = input;
+      const descendantIds = await getDescendantIds(sectionId, basedb);
+      const sectionSchedules = sectionQuery.filter(section => section.sectionId === sectionId);
+      const descendantSectionSchedules = sectionQuery
+        .filter(section => descendantIds.includes(section.sectionId))
+        .map(section => section.start);
```
Comment (solocommand) [inline services/graphql-server/src/graphql/resolvers/platform/content.js]: I wasn't explicit about it in my review, but combine these. There's no need to create two arrays just to join them again later. The website section ID is always included in its own descendant ids, but you can also just include it in your ids to check.
```js
const ids = [
  sectionId,
  ...(useSectionBubbling ? await getDescendantIds(...) : []),
];

**Comment by zarathustra323** [comment]
@Shinsina we're going to hold on these changes for now, pending some other options

**Pull request closed without merging**